### PR TITLE
Allow duplicate headers

### PR DESCRIPTION
**Pull request by jsermeno**
Issue: https://github.com/haskell-servant/servant/issues/396

### CODE REVIEW

**Comment by jkarni** [comment]
Should we just remove the `Contains` check altogether? `Set-Cookie` is not the only header for which this makes sense.

**Comment by jsermeno** [comment]
That works for me. It's a simple immediate solution that allows the most flexibility.

**Comment by jkarni** [comment]
LGTM.

File: servant/src/Servant/API/ResponseHeaders.hs [hunk 1]
```diff
@@ -112,7 +111,7 @@ class AddHeader h v orig new
   addHeader :: v -> orig -> new  -- ^ N.B.: The same header can't be added multiple times
 
 
-instance OVERLAPPING_ ( KnownSymbol h, ToByteString v, Contains h (fst ': rest) ~ 'False)
+instance OVERLAPPING_ ( KnownSymbol h, ToByteString v)
```
Comment (3noch) [inline servant/src/Servant/API/ResponseHeaders.hs]: Inconsistent spacing (no space before `)`). Here and above.

**Comment by jsermeno** [comment]
Fixed, rebased, and I also edited the whitespace of a few other lines for consistency.

**Comment by soenkehahn** [comment]
@jsermeno: This generally allows duplicate headers, right?

LGTM.

**Comment by jsermeno** [comment]
Right, it allows all duplicate headers.

**Comment by soenkehahn** [comment]
@jsermeno: Thanks, was just confused by the title.

**Comment by jkarni** [comment]
Thanks!

**Pull request merged**